File: 7-kyu/letterbox-paint-squad/c++/solution.cpp
#include <array>
#include <vector>

using namespace std;

vector<int> get_digits(int n)
{
    vector<int> ret;
    while (n > 0) {
        ret.push_back(n % 10);
        n /= 10;
    }
    return ret;
}

array<int, 10> paint_letterboxes(int start, int end)
{
    array<int, 10> ret = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
    for (int i = start; i <= end; i++) {
        auto digit_vec = get_digits(i);
        for (const auto &n : digit_vec) {
            ret[n]++;
        }
    }
    return ret;
}

File: 7-kyu/letterbox-paint-squad/javascript/solution.js
const paintLetterboxes = (start, end) => {
  let ret = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
  for (let i = start; i <= end; i++) {
    const iStr = i.toString();
    iStr.split("").forEach(elem => ret[parseInt(elem)] += 1);
  }
  return ret;
};
